src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java

```diff
@@ -62,4 +70,31 @@ public void testFindAssociatedNotLinkedFilesForEmptySearchDir() throws Exception
         List<LinkedFile> actual = util.findAssociatedNotLinkedFiles(entry);
         assertEquals(Collections.emptyList(), actual);
     }
+
+    @Test
+    public void testFileLinksAfterMoving() throws Exception {
+        // Run "Automatically set file links" - check that the bib file was not modified
+        LinkedFile linkedFile = new LinkedFile("desc", path, "PDF");
+        List<LinkedFile> listLinked = new ArrayList<>();
+        listLinked.add(linkedFile);
+        entry.setFiles(listLinked);
+
+        // Copy Bib file from A to B
+        Path destination = B.resolve("CiteKey.pdf");
+        Files.copy(A.resolve("CiteKey.pdf"), destination, StandardCopyOption.REPLACE_EXISTING);
+        Files.deleteIfExists(A.resolve("CiteKey.pdf"));
```

Comment:
There should be another test where the file stays the same.

In test code, it is very OK to have duplicated code fragments.